toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/AuthenticatorState.kt

```diff
@@ -418,30 +419,31 @@ private class AuthenticatorStateImpl(
      * @return [NetworkAuthenticationChallengeResponse] based on user response.
      * @since 200.2.0
      */
-    private suspend fun awaitServerTrustChallengeResponse(networkAuthenticationChallenge: NetworkAuthenticationChallenge): NetworkAuthenticationChallengeResponse =
-        suspendCancellableCoroutine { continuation ->
-            _pendingServerTrustChallenge.value =
-                ServerTrustChallenge(networkAuthenticationChallenge) { shouldTrustServer ->
+    private suspend fun awaitServerTrustChallengeResponse(
+        networkAuthenticationChallenge: NetworkAuthenticationChallenge
+    ): NetworkAuthenticationChallengeResponse {
+        val response = safeSuspendableCancellable<Boolean?> { safeContinuation ->
```

Comment:
As above, I see a warning here about redundant type argument. Also suggest to name `response` variable to something more descriptive:
```suggestion
        val shouldTrustServer = safeSuspendableCancellable { safeContinuation ->
```